homeassistant/components/persistent_notification/__init__.py

```diff
@@ -28,6 +30,8 @@
 
 EVENT_PERSISTENT_NOTIFICATIONS_UPDATED = "persistent_notifications_updated"
 
+NOTIFY_SERVICE_NAME = DOMAIN
```

Comment:
We need to ensure that no other platform registers this notifier name before this platform. Maybe by adding a check in the notify integration for this name. That's not ideal from an architecture view but I'm not sure there's a better way at the moment.

Reply:
Yeah. I’m not really a fan of adding such a check. I’d sooner just have the notify integration load this platform directly in its own setup method before it processes the platforms in the yaml. While weird, that feels like less of an architectural violation to me. It is equivalent to the notify platform just adding a hard coded record to the front of its yaml config before processing it, and like all integrations, notify is allowed to know that persistent_notification exists.

Alternatively, instead of hard coding, we could force or allow the user to configure this, and thus specify a name of their choosing, and make it their responsibility to avoid conflicts. If we go that way, the question is how.

We could just let the user specify this by adding it under the notify integration yaml key, in the typical fashion. While it violates the intent of [ADR0007](https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md), it does not technically violate the terms, since both are existing integrations, and no change to the yaml schema is requires, as the existing base schema from notify is all that would be needed.

We could alternatively follow the spirit of ADR0007, and let this be configured as part of the schema for the persistent_notification platform. This could allow us to keep a default name, but allow the user to override it. This feels kind of pointless, as there is no real need to be able to change the name, except to avoid issues if something else uses that name first. However, it is architecturally the cleanest solution to the potential for name conflicts.

What we cannot do is configure this via config flow, as it does not integrate with a device or external service, so [ADR0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) mandates that this be configured by either yaml or storage collections.

Reply:
Loading this platform first, before other platforms, in the notify integration sounds ok.

Reply:
Ok. Changed to load it first by prepending it to the list of platform to load that got pulled from YAML.